doc/developers/advanced_installation.rst

```diff
@@ -102,7 +102,9 @@ runtime:
 - threadpoolctl (>= |ThreadpoolctlMinVersion|).
 
 Those dependencies are **automatically installed by pip** if they were missing
-when building scikit-learn from source.
+when building scikit-learn from source using `pip install -e .`. If
+`--no-build-isolation` flag is provided, then NumPy and SciPy needs to be
```

Comment:
now that we have the `pyproject.toml` file, can anyone even  pip install without using the `--no-build-isolation` flag?

Reply:
`pyproject.toml` is what enables `pip install -e .` to work from a clean environment.

```bash
python -m venv venv
source venv/bin/activate
pip install -e .

pip list | grep scikit-learn
# scikit-learn  1.1.dev0 FILE_LOCATION
```

Although rebuilds are slow because the Cython/C needs to be rebuilt again. I think `pip install -e .` is better for a user that wants to build and use the project, because it works on a clean environment.

`--no-build-isolation` is better for contributors/developers that want to work on the project itself, which may need to rebuilt due to changes in Cython/C files.

With that in mind, maybe it's just better to remove the statement.